packages/@webex/internal-plugin-metrics/src/types.ts

```diff
@@ -0,0 +1,62 @@
+import {ClientEvent as RawClientEvent} from './call-diagnostic/types/ClientEvent';
+import {FeatureEvent as RawFeatureEvent} from './call-diagnostic/types/FeatureEvent';
+import {MediaQualityEvent as RawMediaQualityEvent} from './call-diagnostic/types/MediaQualityEvent';
+import {SubmitClientEventOptions} from './call-diagnostic/call-diagnostic-metrics';
+
+export type InternalEvent = {
+  name: 'client.pageJMT.received';
+  payload?: RawClientEvent;
+  options: SubmitClientEventOptions;
+};
+
+interface GenericMetric {
+  name: string;
+  payload?: any;
+  options?: any;
+}
+
+export interface ClientEvent extends GenericMetric {
+  name: RawClientEvent['name'];
+  payload?: RawClientEvent;
+  options: SubmitClientEventOptions;
```

Comment:
ideally you should have separated definition for each event:
```ts
interface ClientFeatureAudioNoiseRemoval {
  name:"client.feature.audio.noise.removal",
  payload: ClientFeatureAudioNoiseRemovalPayload
}

interface ClientFeatureEmbedObjectInfo {
  name:"client.feature.embedded-object-info",
  payload: ClientFeatureEmbedObjectInfoPayload
}

interface ClientFeatureGestureRecognition {
  name:"client.feature.embedded-object-info",
  options: ClientFeatureGestureRecognitionOptions
}

// ...


type ClientEvent = 
  | ClientFeatureAudioNoiseRemoval 
  | ClientFeatureEmbedObjectInfo 
  | ClientFeatureGestureRecognition
  | ...
```
some for the other events